File: wanted-market-api/src/main/java/wanted/market/api/domain/product/entity/Product.java
package wanted.market.api.domain.product.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import wanted.market.api.domain.product.enums.ProductStatus;
import wanted.market.api.domain.user.entity.User;

import java.time.LocalDateTime;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
@Table(name= "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id")
    private Long id;
    @Column(name = "product_name")
    private String name;
    @Column
    private Long price;
    @Column
    private Long count;
    @Column
    @Enumerated(EnumType.STRING)
    private ProductStatus productStatus;
    @Column
    private LocalDateTime registerTime;
    @ManyToOne
    @JoinColumn(name ="seller_id")
    private User user;


    @Builder
    public Product(String name, Long price, Long count, User user){
        this.name = name;
        this.price = price;
        this.count = count;
        this.user = user;
        this.productStatus = ProductStatus.SALE;
        this.registerTime = LocalDateTime.now();
    }

    public void modifiedPrice(Long price){
        this.price = price;
    }
}

File: src/main/kotlin/com/api/jellomarket/domain/product/Product.kt
package com.api.jellomarket.domain.product

import com.api.jellomarket.enums.product.ProductState
import jakarta.persistence.*

@Entity
@Table(name = "jello_product")
class Product (
    @Id
    @Column(name = "product_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var productId: Long? = null,

    var name: String,

    var price: Int,

    var state: ProductState,

    var stock: Int,

    var description: String? = null,

    @Column(name = "image_url")
    var imageUrl: String? = null,

    @Column(name = "seller_id")
    var sellerId: Long? = null,

    @Column(name = "created_at")
    var createdAt: String? = null,

    @Column(name = "updated_at")
    var updatedAt: String? = null,

    @Column(name = "deleted_at")
    var deletedAt: String? = null
) {

}